### PR TITLE
Clean up dependencies and devDependencies

### PR DESCRIPTION
**Pull request by alex-page**
Closes https://github.com/Shopify/polaris/issues/231

- [x] Dependencies are required to run
- [x] devDependencies only to develop, e.g.: unit tests, CoffeeScript to JavaScript transpilation, minification, ...

### CODE REVIEW

**Review by samrose3 — APPROVED** [review]
+1 to Kyle's comments but also `vite` and `@vanilla-extract/vite-plugin`.

File: package.json [hunk 1]
```diff
@@ -21,33 +21,28 @@
...
     "size-limit": "^5.0.2",
     "typescript": "^4.3.2",
-    "vite": "^2.4.4",
     "vite-jest": "^0.0.3"
-  },
-  "prettier": {
-    "arrowParens": "always",
-    "singleQuote": true,
-    "trailingComma": "all",
-    "bracketSpacing": false
```
Comment (alex-page) [inline package.json:51]: Should not be needed as we have
https://github.com/Shopify/polaris/blob/7915a869bd9e866d66be90e967b56621209e40a4/.eslintrc#L8

Comment (alex-page) [reply package.json:51]: @kyledurand @samrose3 do you see any issues with this change?

Comment (kyledurand) [reply package.json:51]: Those should all be dupes from the shopify plugin

**Comment by samrose3** [comment]
Just a note, looks like a "Shopify" way of handling deps in a monorepo is to put them all in `devDeps` at the root level ([sewing-kit](https://github.com/Shopify/sewing-kit/blob/main/package.json#L82) and [quilt](https://github.com/Shopify/quilt/blob/main/package.json#L36) examples)

I'm still indifferent! Just thought I'd mention the observation 👀

**Pull request merged**